packages/sync-service/lib/electric/plug/serve_shape_plug.ex

```
@@ -330,8 +330,14 @@ defmodule Electric.Plug.ServeShapePlug do
         {:ok, conn} ->
           {:cont, conn}
 
+        # The documented way to handle closed connections
+        # See https://hexdocs.pm/plug/Plug.Conn.html#chunk/2
```

Technically, the documented return type is `{:error, term}`, so Plug does not restrict what can be returned from `chunk/2`. I believe that the `{:error, :closed}` tuple in the example code there is either valid for the Cowboy adapter (which was the first one used with Plug historically) or is simply an illustrative example.

Regardless, we should be more liberal in the types of error reasons we accept here. Let's just match against any term and log it as an error.

For the curious, since we're using Bandit instead of Cowboy, here's the underlying implemenation of `chunk` that gets called - https://github.com/mtrudel/bandit/blob/6dd82d9625ddce43685f91f8ab5a3150e1fd6a61/lib/bandit/adapter.ex#L190-L205.

---

Ah, make sense! I might get rid of the `:closed` case then.

As for handling other cases, what's wrong with let it crash?

---

I believe https://github.com/electric-sql/electric/issues/1440 was raised as an issue to begin with because just letting it crash resulted in a hard-to-read error that seemed to point to a bug in the server.

To fix that, we can match on `{:error, reason}` and log the reason in a way that won't confuse us when we see it in the logs.

---

Ok, good point. It would make for an easier to read error message. I'll update it.

---

Ah, I think you meant that if there's an error with a reason other than `"closed`", we should let it crash to see if it's pointing at an actual bug?

Fair enough. Though I would love to have some explanation for this `"closed"` reason we're handling in the code. Could be a URL to this issue.